src/bot.py

```diff
@@ -136,48 +136,23 @@ async def stale_queue_timer():
 @client.command(name='q', aliases=['addmepapanorm', 'Q', 'addmebitch', 'queue', 'join'], pass_context=True)
 async def q(ctx, *arg):
     players = getBallChaserList()
-    messages = SixMans.playerQueue(
+    response = SixMans.playerQueue(
         ctx.message.author,
         REPORT_CH_IDS[0] if (len(REPORT_CH_IDS) > 0) else -1,
         *arg,
     )
     author = ctx.message.author
-    for msg in messages:
-        if (isinstance(msg, Embed)):
-            if (getActiveMatch(author) is not None):
-                for i in players:
-                    user = await client.fetch_user(str(i.id))
-                    await user.send(embed=msg)
-                
-                await author.send(embed=msg)
+    if (response.sendPrivately):
+        async def sendPrivateMsg(playerId: str):
+            user = await client.fetch_user(playerId)
+            if (not user.bot):
+                await user.send(embed=response.embed)
```

Comment:
I like this a lot. I noticed that Norm took some time to push out DM's to each person before posting another message in the channel.